### PR TITLE
ag grepper: New package

### PR DESCRIPTION
**Pull request by forgottenswitch**
"A better grep" and an addon package.
Distributions name it the_silver_searcher.

### CODE REVIEW

File: ag/PKGBUILD [hunk 1]
```diff
@@ -0,0 +1,33 @@
+# Maintainer: Mihail Konev <k.mvc@ya.ru>
```
Comment (Chocobo1) [inline ag/PKGBUILD]: IMO this folder should be named `ag-git`. 
There should be another folder `ag` which installs stable release version.

The same applies to `cgvg`.

File: ag/PKGBUILD [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+pkgname=ag
+srcname=the_silver_searcher
+pkgver=git
```
Comment (Chocobo1) [inline ag/PKGBUILD]: A fixed `pkgver` disables the ability to update this package (via pacman) in the future.
See https://aur.archlinux.org/packages/?O=0&K=-git for a better `pkgver` scheme.

Also if you have time: 
https://git.archlinux.org/svntogit/community.git/tree/trunk/PKGBUILD?h=packages/the_silver_searcher
https://wiki.archlinux.org/index.php/Arch_packaging_standards
https://wiki.archlinux.org/index.php/PKGBUILD

**Comment by mati865** [comment]
`pkgver` is wrong. For git packages it should use commit number like here https://github.com/Alexpux/MSYS2-packages/blob/master/gyp-git/PKGBUILD

**Comment by forgottenswitch** [comment]
This cgvg currently does not have config file, like the original ...
https://github.com/uzi/cgvg/blob/master/cgvgrc-example

**Comment by mati865** [comment]
`pkgver` is still wrong, look here: https://github.com/Alexpux/MSYS2-packages/blob/master/gyp-git/PKGBUILD#L24

File: ag-git/PKGBUILD [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+srcname=the_silver_searcher
+pkgver=git
+pkgrel=1
+pkgdesc="A better grep"
+arch=('i686' 'x86_64')
+license=('Apache')
+url="https://github.com/ggreer/the_silver_searcher"
+groups=('base' 'base-devel')
+provides=('ag')
+conflicts=('ag')
```
Comment (mati865) [inline ag-git/PKGBUILD]: Remove `provides=('ag')`, `conflicts=('ag')` and `groups=('base' 'base-devel')`

File: ag-git/PKGBUILD [hunk 4]
```diff
@@ -0,0 +1,35 @@
+# Maintainer: Mihail Konev <k.mvc@ya.ru>
+
+pkgname=ag
```
Comment (mati865) [inline ag-git/PKGBUILD]: change `pkgname=ag` to `pkgname=ag-git`

File: cgvg-git/PKGBUILD [hunk 5]
```diff
@@ -0,0 +1,29 @@
+# Maintainer: Mihail Konev <k.mvc@ya.ru>
+
+pkgname=cgvg
```
Comment (mati865) [inline cgvg-git/PKGBUILD]: change `pkgname=cgvg` to `pkgname=cgvg-git`

File: cgvg-git/PKGBUILD [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+pkgver=git
+pkgrel=1
+pkgdesc="Grep with numerical goto"
+arch=('any')
+license=('MIT')
+url="https://github.com/forgottenswitch/cgvg"
+groups=('base' 'base-devel')
+provides=('cgvg')
+conflicts=('cgvg')
```
Comment (mati865) [inline cgvg-git/PKGBUILD]: remove `groups=('base' 'base-devel')`, `provides=('cgvg')`, `conflicts=('cgvg')`

File: ag-git/PKGBUILD (hunk 3 above)
Comment (forgottenswitch) [inline ag-git/PKGBUILD]: https://github.com/Alexpux/MSYS2-packages/blob/master/mintty-git/PKGBUILD#L11

**Comment by forgottenswitch** [comment]
The cgvg is also implement here:
https://github.com/sampson-chen/sack

**Comment by mingwandroid** [comment]
We already have ag in MINGW-packages so I'm against adding it here too. This came up already about a year ago.

**Comment by forgottenswitch** [comment]
But in mingw it clashes with stdin being emulated with named pipes.

**Comment by forgottenswitch** [comment]
There indeed...

**Comment by mingwandroid** [comment]
The named pipe issue can be fixed with some tests for the pipe names. Git for Windows does this.

Slightly related, until the pipe handling is fixed, to prevent MINGW-packages ag from thinking it's in pipe mode pass `.` as an argument to indicate the current working directory.

**Comment by forgottenswitch** [comment]
Put this into header and patch ag, nodejs, ffmpeg, everything `stdin is not a tty`?
https://github.com/git-for-windows/git/pull/102/commits

**Comment by mingwandroid** [comment]
Yes, but we need to be careful of licenses. @kblees, would you consider relicensing this code under BSD so we can use it in GPL and BSD software in MSYS2? If not then we'll need to rewrite it.

**Comment by forgottenswitch** [comment]
As long as we have this grep, no need to copypaste.

**Comment by mingwandroid** [comment]
Sure.

**Comment by forgottenswitch** [comment]
Minimal working example:
https://github.com/forgottenswitch/mingw-isatty

**Comment by forgottenswitch** [comment]
Didn't tested it properly, just copied `iscygtty.c` to `/mingw64/x86_64.../include`,
and built ag from source. Tty detection worked.
https://github.com/Alexpux/mingw-w64/compare/master...forgottenswitch:cygtty
https://github.com/Alexpux/MINGW-packages/compare/master...forgottenswitch:ag-tty
Ag patch was formatted from:
https://github.com/ggreer/the_silver_searcher/compare/master...forgottenswitch:cygtty

**Pull request closed without merging**